tuf/api/metadata.py

```diff
@@ -494,32 +535,95 @@ def __init__(
             self, _type: str, version: int, spec_version: str,
             expires: datetime, meta: JsonDict) -> None:
         super().__init__(_type, version, spec_version, expires)
-        # TODO: Add class for meta
-        self.meta = meta
+
+        self.meta = {}
+        for meta_path, meta_info in meta.items():
+            self.meta[meta_path] = MetaFile(meta_info['version'],
+                meta_info.get('length'), meta_info.get('hashes'))
+
 
     # Serialization.
     def to_dict(self) -> JsonDict:
         """Returns the JSON-serializable dictionary representation of self. """
         json_dict = super().to_dict()
+        meta_dict = {}
+        for meta_path, meta_info in self.meta.items():
+            meta_dict[meta_path] = meta_info.to_dict()
+
         json_dict.update({
-            'meta': self.meta
+            'meta': meta_dict
         })
         return json_dict
 
 
     # Modification.
-    def update(
+    def update_meta_file(
             self, rolename: str, version: int, length: Optional[int] = None,
             hashes: Optional[JsonDict] = None) -> None:
         """Assigns passed (delegated) targets role info to meta dict. """
         metadata_fn = f'{rolename}.json'
 
-        self.meta[metadata_fn] = {'version': version}
-        if length is not None:
-            self.meta[metadata_fn]['length'] = length
+        self.meta[metadata_fn] = MetaFile(version, length, hashes)
+
+
+class TargetFile:
```

Comment:
What to you think about `TargetFileInfo` akin to my suggestion `MetadataFileInfo` above?